src/OrchardCore/OrchardCore.Data.YesSql/Removing/ShellDbTablesInfo.cs

```diff
@@ -0,0 +1,331 @@
+using System;
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Linq;
+using Dapper;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using YesSql;
+using YesSql.Services;
+using YesSql.Sql;
+using YesSql.Sql.Schema;
+
+namespace OrchardCore.Environment.Shell.Removing;
+
+internal class ShellDbTablesInfo : ISchemaBuilder
+{
+    private ICommandInterpreter _commandInterpreter;
+    public string DatabaseProvider { get; set; }
+    public string TablePrefix { get; set; }
+    public IEnumerable<string> TableNames { get; set; } = Enumerable.Empty<string>();
+    public ISqlDialect Dialect { get; private set; }
+    public ITableNameConvention TableNameConvention { get; private set; }
+    public DbConnection Connection { get; set; }
+    public DbTransaction Transaction { get; set; }
+    public bool ThrowOnError { get; private set; }
+    public ILogger _logger { get; set; } = NullLogger.Instance;
+
+    public HashSet<(string Name, Type Type, string Collection)> MapIndexTables { get; private set; } =
+        new HashSet<(string Name, Type Type, string Collection)>();
+
+    public HashSet<(string Name, Type Type, string Collection)> ReduceIndexTables { get; private set; } =
+        new HashSet<(string Name, Type Type, string Collection)>();
+
+    public HashSet<string> BridgeTables { get; private set; } = new HashSet<string>();
+    public HashSet<string> DocumentTables { get; private set; } = new HashSet<string>();
+    public HashSet<string> Tables { get; private set; } = new HashSet<string>();
+
+    public ShellDbTablesInfo Configure(ShellSettings shellSettings)
+    {
+        DatabaseProvider = shellSettings["DatabaseProvider"];
+
```

Comment:
I think here you also need `TablePrefix = shellSettings["TablePrefixSeparator"] ?? "_";`

Reply:
Will look at it, did you introduce it? Didn't see yet its usage on YesSql side.

Reply:
This is part of OC not YesSql. We introduced it about a month ago

Reply:
Here we already use `TablePrefix` from the YesSql configuration, so already including the `TablePrefixSeparator`.

Reply:
good!